### PR TITLE
fix shader compilation error and invisible color layers in certain circumstances

### LINKED ISSUE #4553: Non-segmentation layers are not rendered
For some of our datasets, containing 5 uint8 color layers and 1 uint32 segmentation layer, only the segmentation layer is rendered for me, regardless of which other layers I activate or deactivate.

See this dataset for example: https://webknossos.brain.mpg.de/annotations/Explorational/5e984d80010000bd629ec9dc#6656,4608,3072,0,2.000

Notes:
- Toggling the segmentation layer leads to a shader error, complaining about a redefinition of some variables related to the segmentation layer in the shader code. This could be a lead for the cause of this bug.
  ```
  ERROR: 0:145: 'layer_c2VnbWVudGF0aW9u_textures' : redefinition
  ERROR: 0:146: 'layer_c2VnbWVudGF0aW9u_lookup_texture' : redefinition
  ERROR: 0:147: 'layer_c2VnbWVudGF0aW9u_data_texture_width' : redefinition
  ERROR: 0:148: 'layer_c2VnbWVudGF0aW9u_maxZoomStep' : redefinition
  ERROR: 0:149: 'layer_c2VnbWVudGF0aW9u_alpha' : redefinition
  ERROR: 0:538: 'getRgbaAtXYIndex_layer_c2VnbWVudGF0aW9u' : function already has a body
  ```
- Removing the segmentation layer from the datasource-properties, fixes the issue and all layers (except for the removed segmentation layer) are displayed.

### PR DESCRIPTION
**Pull request by philippotto**
This PR fixes two bugs (both mentioned in #4553).

One issue was a `===` comparison between a boolean and undefined (I thought, the flow linter would warn about this, but unfortunately it doesn't. we disabled sketchy-bool checks because it also warns about a lot of other stuff, which is usually fine for is).

The other issue was that the segmentation layer name could slip into the `colorLayerNames` variable. The reason for this was that the LRU layer could contain the segmentation layer. I made the naming clearer and excluded the segmentation layer from the LRU system (since the shader s always compiled for thesegmentation layer; one could re-think this, but that's outside the scope of this PR imo).

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- I created a dataset which contains too many layers to render all at once and a segmentation layer
- I also ensured that toggling the segmentation layer doesn't throw an error anymore and that the color layer can now be rendered.
I think the case in question doesn't need to be tested again.

### Issues:
- fixes #4553

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/geometries/materials/plane_material_factory.js [hunk 1]
```diff
@@ -605,32 +606,39 @@ class PlaneMaterialFactory {
...
-      ...this.leastRecentlyVisibleLayers,
-      ..._.without(disabledLayerNames, ...this.leastRecentlyVisibleLayers),
+    this.leastRecentlyVisibleColorLayers = [
+      ...this.leastRecentlyVisibleColorLayers,
+      ..._.without(disabledLayerNames, ...this.leastRecentlyVisibleColorLayers),
     ];
 
-    return enabledLayerNames
-      .concat(this.leastRecentlyVisibleLayers)
+    const names = enabledLayerNames
```
Comment (philippotto) [inline frontend/javascripts/oxalis/geometries/materials/plane_material_factory.js:619]: this change makes debugging a bit easier.

File: frontend/javascripts/oxalis/view/settings/dataset_settings_view.js [hunk 2]
```diff
@@ -224,7 +224,7 @@ class DatasetSettings extends React.PureComponent<DatasetSettingsProps> {
           <span style={{ fontWeight: 700 }}>
             {!isColorLayer && isVolumeTracing ? "Volume Layer" : layerName}
           </span>
-          <Tag style={{ cursor: "default", marginLeft: 8 }}>{elementClass} Layer</Tag>
+          <Tag style={{ cursor: "default", marginLeft: 8 }}>{elementClass}</Tag>
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/settings/dataset_settings_view.js:227]: I think the `Layer` label is not really necessary.

File: frontend/javascripts/oxalis/geometries/materials/plane_material_factory.js [hunk 3]
```diff
@@ -605,32 +606,39 @@ class PlaneMaterialFactory {
...
-      .map(sanitizeName);
+      .sort();
+
+    return names.map(sanitizeName);
   }
 
-  onDisableLayer = layerName => {
-    this.leastRecentlyVisibleLayers = _.without(this.leastRecentlyVisibleLayers, layerName);
-    this.leastRecentlyVisibleLayers = [layerName, ...this.leastRecentlyVisibleLayers];
+  onDisableColorLayer = (layerName, isSegmentationLayer) => {
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/geometries/materials/plane_material_factory.js:627]:
```suggestion
  onDisableColorLayer = layerName => {
```

File: frontend/javascripts/oxalis/geometries/materials/plane_material_factory.js [hunk 4]
```diff
@@ -605,32 +606,39 @@ class PlaneMaterialFactory {
...
+      layerName,
+    );
+    this.leastRecentlyVisibleColorLayers = [layerName, ...this.leastRecentlyVisibleColorLayers];
 
     this.recomputeFragmentShader();
   };
 
-  onEnableLayer = layerName => {
-    this.leastRecentlyVisibleLayers = _.without(this.leastRecentlyVisibleLayers, layerName);
+  onEnableColorLayer = (layerName, isSegmentationLayer) => {
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/geometries/materials/plane_material_factory.js:637]:
```suggestion
  onEnableColorLayer = layerName => {
```

**Pull request merged**